code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/OptimizeExtension.java

```diff
@@ -61,11 +62,19 @@ public boolean doWork(final Event event) {
 
     // Map containing the update event IDs (and corresponding requested scopes) for Edge events that
     // haven't yet received an Edge completion response.
-    private Map<String, List<DecisionScope>> updateRequestEventIdsInProgress = new HashMap<>();
+    // Concurrent Map containing the update event IDs (and corresponding requested scopes) for Edge
+    // events that haven't yet received an Edge completion response.
+    // This is accessed from multiple threads.
+    private final Map<String, List<DecisionScope>> updateRequestEventIdsInProgress =
+            new ConcurrentHashMap<>();
 
     // a dictionary to accumulate propositions returned in various personalization:decisions events
     // for the same Edge personalization request.
-    private Map<DecisionScope, OptimizeProposition> propositionsInProgress = new HashMap<>();
+    // Concurrent Map containing the get propositions (and corresponding requested scopes) for Edge
+    // events that are going to be pushed to the Edge network.
+    // This is accessed from multiple threads.
```

Comment:
Can you replace line 71 - 75 with 

    // Concurrent Map to accumulate propositions returned in various personalization:decisions events
    // for the same Edge personalization request.
    // This is accessed from multiple threads.

Reply:
done @praveek